independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -541,7 +541,15 @@ public boolean test(MethodInfo method) {
                         continue; // hope for the best
                     }
                     if (Modifier.isPrivate(parameterClassInfo.flags())) {
-                        return true; // parameters whose class is private can not be loaded, as we would end up with IllegalAccessError when trying to access the use the load the class
+                        if (Modifier.isPrivate(method.flags())) {
+                            return true;
```

Comment:
How does it help? I mean private methods can be intercepted as well (e.g. observer methods)...

Reply:
I believe the intention is to indeed keep supporting `private String someMethod(User user)` but not `public String someMethod(User user)` where `User` is private - which is a problem for the security and other checks - at least if it is a REST endpoint method.

Reply:
I understand, but since `private String someMethod(User user)` can be intercepted as well, we could run into the same problem.

Reply:
Oh, I see. The particular variation which causes the problem with the checks is `public String someMethod(User user)` with User being private. 
Hmm, may be it should be addressed at the the Resteasy/Reactive Resteasy level, when the resource methods are analyzed...

Reply:
Will this actually cause problems for a private interceptor method though? They must be done through a different mechanism that overriding the method?

Reply:
You're right that we invoke private intercepted methods via reflection. But TBH I'm not sure if it matters. Actually, the original "fix" was wrong because something like this should work just fine:
```java
   @ApplicationScoped
   class FooObserver {
        
      @MyInterceptor 
      void observeFoo(@Observes Foo value) {
          //...
        }
    }

    private class Foo {
    }
```
I think that the correct way would be to throw an exception for a private parameter type not defined on the original class...